user_sync/post_sync/connectors/sign_sync/client.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import requests
+import json
+
+
+class SignClient:
+    version = 'v5'
+    _endpoint_template = 'api/rest/{}/'
+    DEFAULT_GROUP_NAME = 'default group'
+
+    def __init__(self, config):
+        for k in ['host', 'key', 'admin_email']:
+            assert k in config, "Key '{}' must be specified for all Sign orgs".format(k)
+        self.host = config['host']
+        self.key = config['key']
+        self.admin_email = config['admin_email']
+        self.console_org = config['console_org'] if 'console_org' in config else None
+        self.api_url = self.base_uri()
+        self.groups = self.get_groups()
+        self.default_group_id, = [_id for name, _id in self.groups.items() if name == self.DEFAULT_GROUP_NAME]
+        self.logger = logging.getLogger(self.logger_name())
+
+    def logger_name(self):
+        return 'sign_client.{}'.format(self.console_org if self.console_org else 'main')
+
+    def header(self):
+        """
+        Return Sign API auth header
+        :return: dict()
+        """
+        if self.version == 'v6':
+            return {
+                "Authorization": "Bearer {}".format(self.key)
+            }
+        return {
+            "Access-Token": self.key
+        }
+
+    def header_json(self):
+        """
+        Get auth headers with options to PUT/POST JSON
+        :return: dict()
+        """
+
+        json_headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+        }
+        json_headers.update(self.header())
+        return json_headers
+
+    def base_uri(self):
+        """
+        This function validates that the SIGN integration key is valid.
+        :return: dict()
+        """
+
+        endpoint = self._endpoint_template.format(self.version)
+        url = 'https://' + self.host + '/' + endpoint
+
+        if self.version == "v6":
+            return requests.get(url + "baseUris", headers=self.header()).json()['apiAccessPoint'] + endpoint
+        return requests.get(url + "base_uris", headers=self.header()).json()['api_access_point'] + endpoint
```

Comment:
We discussed on the call, but error handling in general will greatly improve the usability of the connector.  In this instance, the call to access a dictionary key 'api_access_point' fails with a cryptic KeyError if the api key is incorrect.